tensorflow_addons/callbacks/tests/avg_model_checkpoint_test.py

```diff
@@ -0,0 +1,206 @@
+import pytest
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow_addons.callbacks import AverageModelCheckpoint
+from tensorflow_addons.optimizers import MovingAverage
+import os
+
+try:
+    import h5py
+except ImportError:
+    h5py = None
+
+TRAIN_SAMPLES = 10
+NUM_CLASSES = 2
+INPUT_DIM = 3
+NUM_HIDDEN = 5
+BATCH_SIZE = 5
+EPOCHS = 5
+test_model_filepath = "test_model.h5"
+
+moving_avg = MovingAverage(
+    tf.keras.optimizers.SGD(lr=2.0), sequential_update=True, average_decay=0.5,
+)
```

Comment:
Please move the optimizer in the `get_data_and_model`, tests should not interfer with each other, and as such, they should not modify any global object (in this case an optimizer). The optimizer should be re-created in each test.